crypto/bio/b_print.c

```diff
@@ -368,28 +368,31 @@ fmtstr(char **sbuffer,
     if (value == 0)
         value = "<NULL>";
 
-    strln = strlen(value);
-    if (strln > INT_MAX)
-        strln = INT_MAX;
+    strln = OPENSSL_strnlen(value, max < 0 ? SIZE_MAX : (size_t)max);
+    if (max >= 0 && strln > (size_t)max)
+        strln = (size_t)max;
```

Comment:
I wonder why this `if` is necessary. Surely if `max >= 0` then `strlen` set from the return value of `OPENSSL_strnlen` will be at most `max`, so this `if` will never evaluate to true.